### PR TITLE
Add end of line filled option on Style configurator

### PR DESCRIPTION
**Pull request by Cyrillev91**
Add  "End of line filled" scintilla option in Style configurator dialog box.

eolfilled_styleconfigurator2

Many lexer use eolfilled option to Highlight string not close but not active in npp, in this PR active this for python.

I think this PR is better than PR #2585 (please confirm), because to activate it should be added **eolfilled = "1"** in "stylers.model.xml" file without changing the code.

**without eolfilled**
pythonlexer_partial

**with eolfilled**
pythonlexer_after

Close issue #2698
Close issue #2584

### CODE REVIEW

File: PowerEditor/src/Parameters.cpp [hunk 1]
```diff
@@ -3334,6 +3334,13 @@ void StyleArray::addStyler(int styleID, TiXmlNode *styleNode)
 			_styleArray[index]._bgColor = (RGB((result >> 16) & 0xFF, (result >> 8) & 0xFF, result & 0xFF)) | (result & 0xFF000000);
 		}
 
+		str = element->Attribute(TEXT("eolfilled"));
+		if (str)
+		{
+			unsigned long result = hexStrVal(str);
```
Comment (MAPJe71) [inline PowerEditor/src/Parameters.cpp]: I guess this line can be removed (remainder of copy-paste).

File: PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp [hunk 2]
```diff
@@ -452,6 +452,9 @@ void ScintillaEditView::setSpecialStyle(const Style & styleToSet)
     if ( styleToSet._colorStyle & COLORSTYLE_BACKGROUND )
 	    execute(SCI_STYLESETBACK, styleID, styleToSet._bgColor);
 
+    if (styleToSet._colorStyle & COLORSTYLE_BACKGROUND)
```
Comment (MAPJe71) [inline PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp]: This if-statement can be combined with the previous if-statement.

File: PowerEditor/src/WinControls/ColourPicker/WordStyleDlg.cpp [hunk 3]
```diff
@@ -747,6 +765,9 @@ void WordStyleDlg::setVisualFromStyleList()
 		_pBgColour->setColour(style._bgColor);
 		_pBgColour->setEnabled((style._colorStyle & COLORSTYLE_BACKGROUND) != 0);
 		isEnable = true;
+		int eolfilled = 0;
+		eolfilled = style._eolFilled;
+		::SendMessage(_hCheckEolFilled, BM_SETCHECK, eolfilled, 0);
 	}
```
Comment (MAPJe71) [inline PowerEditor/src/WinControls/ColourPicker/WordStyleDlg.cpp]: Why not skip the intermediate variable with a single statement like `::SendMessage(_hCheckEolFilled, BM_SETCHECK, (style._eolFilled ? 1 : 0), 0);`?

File: PowerEditor/src/stylers.model.xml [hunk 4]
```diff
@@ -656,6 +656,8 @@
             <WordsStyle name="OPERATOR" styleID="10" fgColor="000080" bgColor="FFFFFF" fontName="" fontStyle="1" fontSize="" />
             <WordsStyle name="IDENTIFIER" styleID="11" fgColor="000000" bgColor="FFFFFF" fontName="" fontStyle="0" fontSize="" />
             <WordsStyle name="COMMENTBLOCK" styleID="12" fgColor="008000" bgColor="FFFFFF" fontName="" fontStyle="0" fontSize="" />
+            <WordsStyle name="STRING NOT CLOSED" styleID="13" fgColor="000000" bgColor="E0C0E0" fontName="" fontStyle="0" fontSize="" eolfilled="1" />
+            <WordsStyle name="USER DEFINED" styleID="14" fgColor="407090" bgColor="FFFFFF" fontName="" fontStyle="1" fontSize="" keywordClass="instre2" />
```
Comment (MAPJe71) [inline PowerEditor/src/stylers.model.xml]: As also/already mentioned in #2585:
1.  I guess separate PR's would have been better.
2.  Furthermore, why add a separate style for "USER DEFINED" keywords when it is already possible to enter user-defined keywords as part of the "KEYWORDS" style?

Comment (RobinHossain2817) [inline PowerEditor/src/stylers.model.xml]: …]()

File: PowerEditor/src/Parameters.cpp (hunk 1 above)
Comment (Cyrillev91) [inline PowerEditor/src/Parameters.cpp]: ok

File: PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp (hunk 2 above)
Comment (Cyrillev91) [inline PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp]: ok

File: PowerEditor/src/WinControls/ColourPicker/WordStyleDlg.cpp (hunk 3 above)
Comment (Cyrillev91) [inline PowerEditor/src/WinControls/ColourPicker/WordStyleDlg.cpp]: ok

File: PowerEditor/src/stylers.model.xml (hunk 4 above)
Comment (Cyrillev91) [inline PowerEditor/src/stylers.model.xml]: Style "USER DEFINE" is in scintilla lexer, but not liked in npp.
It allows to have two differents highlighting for "USER DEFINED" style and "KEYWORDS" style

**Comment by donho** [comment]
@Cyrillev91 
The idea of PR is good. 
However, it doesn't work for a lot of case (for exemple: NSIS comment line).
The thing is, every lexer (for each language)  is coded differently (and maintained by different people), that makes the behaviour of token EOL differently.
Since this feature will bring a lot bug report, I prefer not to offer it.

**Pull request closed without merging**